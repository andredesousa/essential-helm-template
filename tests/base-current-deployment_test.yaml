suite: V1 test deployment
values:
  - ../values.yaml
templates:
  - base-current-deployment.yaml
tests:
  - it: should have 2 documents
    asserts:
      - hasDocuments:
          count: 2
  - it: should have a deployment document
    documentIndex: 0
    asserts:
      - isAPIVersion:
          of: apps/v1
      - isKind:
          of: Deployment
      - isNotEmpty:
          path: metadata
      - isNotEmpty:
          path: spec
      - equal:
          path: spec.replicas
          value: 1
      - equal:
          path: spec.selector.matchLabels.app
          value: aks-current-api
      - matchRegex:
          path: spec.template.spec.containers[0].image
          pattern: mcr.microsoft.com\/azuredocs\/aks-helloworld
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
      - contains:
          path: spec.template.spec.containers[0].ports
          content:
            containerPort: 80
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: TITLE
            value: "Welcome to Azure Kubernetes Service (AKS)"
  - it: should have a service document
    documentIndex: 1
    asserts:
      - isAPIVersion:
          of: v1
      - isKind:
          of: Service
      - isNotEmpty:
          path: metadata
      - isNotEmpty:
          path: spec
      - equal:
          path: spec.selector.app
          value: aks-current-api
      - equal:
          path: spec.type
          value: ClusterIP
      - contains:
          path: spec.ports
          content:
            port: 80
            protocol: TCP
      - matchSnapshot:
          path: spec
