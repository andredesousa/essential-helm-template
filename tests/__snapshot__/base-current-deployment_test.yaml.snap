should pass all kinds of assertion:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app: aks-current-api
    template:
      metadata:
        labels:
          app: aks-current-api
      spec:
        containers:
        - env:
          - name: TITLE
            value: Welcome to Azure Kubernetes Service (AKS)
          image: mcr.microsoft.com/azuredocs/aks-helloworld:v1
          imagePullPolicy: IfNotPresent
          name: aks-current-api
          ports:
          - containerPort: 80
